### ISSUE TITLE
TestConvertToEnvoyJwtConfig test flake

### ISSUE THREAD

**Issue by howardjohn** [istio/istio#46623]
https://storage.googleapis.com/istio-prow/pr-logs/pull/batch/unit-tests_istio/1693653620689473536/build-log.txt

```
==================
WARNING: DATA RACE
Write at 0x00c00096a57f by goroutine 293:
  istio.io/istio/pilot/pkg/model.(*JwksResolver).refresh.func1.1()
      /home/prow/go/src/istio.io/istio/pilot/pkg/model/jwks_resolver.go:469 +0x16a

Previous write at 0x00c00096a57f by goroutine 294:
  istio.io/istio/pilot/pkg/model.(*JwksResolver).refresh.func1.1()
      /home/prow/go/src/istio.io/istio/pilot/pkg/model/jwks_resolver.go:479 +0x38a

Goroutine 293 (running) created at:
  istio.io/istio/pilot/pkg/model.(*JwksResolver).refresh.func1()
      /home/prow/go/src/istio.io/istio/pilot/pkg/model/jwks_resolver.go:461 +0x77e
  sync.(*Map).Range()
      /usr/local/go/src/sync/map.go:476 +0x1db
  istio.io/istio/pilot/pkg/model.(*JwksResolver).refresh()
      /home/prow/go/src/istio.io/istio/pilot/pkg/model/jwks_resolver.go:437 +0xee
  istio.io/istio/pilot/pkg/model.(*JwksResolver).refreshCache()
      /home/prow/go/src/istio.io/istio/pilot/pkg/model/jwks_resolver.go:413 +0x2e
  istio.io/istio/pilot/pkg/model.(*JwksResolver).refresher()
      /home/prow/go/src/istio.io/istio/pilot/pkg/model/jwks_resolver.go:406 +0x22c
  istio.io/istio/pilot/pkg/model.newJwksResolverWithCABundlePaths.func1()
      /home/prow/go/src/istio.io/istio/pilot/pkg/model/jwks_resolver.go:222 +0x33

Goroutine 294 (finished) created at:
  istio.io/istio/pilot/pkg/model.(*JwksResolver).refresh.func1()
      /home/prow/go/src/istio.io/istio/pilot/pkg/model/jwks_resolver.go:461 +0x77e
  sync.(*Map).Range()
      /usr/local/go/src/sync/map.go:476 +0x1db
  istio.io/istio/pilot/pkg/model.(*JwksResolver).refresh()
      /home/prow/go/src/istio.io/istio/pilot/pkg/model/jwks_resolver.go:437 +0xee
  istio.io/istio/pilot/pkg/model.(*JwksResolver).refreshCache()
      /home/prow/go/src/istio.io/istio/pilot/pkg/model/jwks_resolver.go:413 +0x2e
  istio.io/istio/pilot/pkg/model.(*JwksResolver).refresher()
      /home/prow/go/src/istio.io/istio/pilot/pkg/model/jwks_resolver.go:406 +0x22c
  istio.io/istio/pilot/pkg/model.newJwksResolverWithCABundlePaths.func1()
      /home/prow/go/src/istio.io/istio/pilot/pkg/model/jwks_resolver.go:222 +0x33
==================
```

Bizarre. No clue how this can happen, and I haven't reproduced it yet

**Comment by howardjohn** [comment]
actually I can reproduce, just takes a looong time

**Comment by howardjohn** [comment]
Actually its pretty obvious, I just read the code wrong. The refresh() spawns multiple goroutines and those are racy

**Comment by howardjohn** [comment]
cc @aryan16

**Comment by wulianglongrd** [comment]
fixed by #46634
